gateway.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Gulp Common - Microsoft Sample Code - Copyright (c) 2016 - Licensed MIT
+ */
+'use strict';
+
+var fs = require('fs');
+var args = require('get-gulp-args')();
+
+var all;
+
+function initTasks(gulp, options) {
+  all = require('./all.js')(options);
+
+  var config = all.getConfig();
+  var workspace = './ble_sample/';
+
+  // stick config into gulp object
+  gulp.config = flatten(config);
+
+  if (typeof all.gulpTaskBI === 'function') {
+    all.gulpTaskBI(gulp, 'nodejs', 'gateway', ((options && options.appName) ? options.appName : 'unknown'));
+  }
+
+  var runSequence = require('run-sequence').use(gulp);
+
+  // copy files into profile folder
+  gulp.task('init', 'Initializes sample', function(cb) {
+
+    if (options.configPostfix && options.configTemplate) {
+      all.updateGlobalConfig(options.configPostfix, options.configTemplate['ssh-config']);
+      saveConfigFile(config.bleConfig, options.configTemplate['ble-config']);
+      saveConfigFile(config.azFuncConfig, options.configTemplate['azure-function-config']);
+    }
+
+    cb();
+  });
+
+  // copy inital files to the NUC
+  gulp.task('setup-remote', 'Copy script to remote', function(cb) {
+    var cpList = [
+      'app/.ble_gateway.json',
+      'app/sensortagdisco.js',
+      'app/testconnect.js',
+      'app/deploy.js',
+      'app/run.js',
+      'app/lib/bleconfig.js',
+      'app/lib/bluetoothctl.js',
+      'app/lib/interactcli.js',
+      'app/lib/util.js'
+    ];
+    var link = [];
```

Comment:
That's a way, but I think this list can be more configurable later. But this time, it just be fixed and only in the function's scope